packages/image_picker/image_picker_for_web/lib/image_picker_for_web.dart

```diff
@@ -226,65 +230,63 @@ class ImagePickerPlugin extends ImagePickerPlatform {
     return null;
   }
 
-  List<html.File>? _getFilesFromInput(html.FileUploadInputElement input) {
+  List<web.File>? _getFilesFromInput(web.HTMLInputElement input) {
     if (_hasOverrides) {
       return _overrides!.getMultipleFilesFromInput(input);
     }
-    return input.files;
+    return input.files?.toList;
   }
 
   /// Handles the OnChange event from a FileUploadInputElement object
   /// Returns a list of selected files.
-  List<html.File>? _handleOnChangeEvent(html.Event event) {
-    final html.FileUploadInputElement? input =
-        event.target as html.FileUploadInputElement?;
+  List<web.File>? _handleOnChangeEvent(web.Event event) {
+    final web.HTMLInputElement? input = event.target as web.HTMLInputElement?;
     return input == null ? null : _getFilesFromInput(input);
   }
 
   /// Monitors an <input type="file"> and returns the selected file(s).
-  Future<List<XFile>> _getSelectedXFiles(html.FileUploadInputElement input) {
+  Future<List<XFile>> _getSelectedXFiles(web.HTMLInputElement input) {
     final Completer<List<XFile>> completer = Completer<List<XFile>>();
     // Observe the input until we can return something
-    input.onChange.first.then((html.Event event) {
-      final List<html.File>? files = _handleOnChangeEvent(event);
+    input.onchange = (web.Event event) {
```

Comment:
Use the streams, not the listeners directly.

Reply:
Created an issue for this, the `onCancel` stream doesn't exist and seems that cannot be easily created: https://github.com/dart-lang/web/issues/199